src/components/Table/GridTable.test.tsx

```diff
@@ -710,6 +710,32 @@ describe("GridTable", () => {
         ),
       ).rejects.toThrow("Column 0 passed an unsortable value, use GridCellContent or clientSideSort=false");
     });
+
+    it("can pin rows and filter - filterable pin", async () => {
+      // Given the table with 3 rows with pins
+      const r = await render(
+        <GridTable<Row>
+          filter={"d"}
+          columns={[nameColumn, valueColumn]}
+          sorting={{ on: "client", initial: [1, "ASC"] }}
```

Comment:
Thought: given we're sorting, I think technically the "name: a" row with a `value: 11` could actually be the 3rd row right now and our assertions below would not catch it?

Kinda wonder, maybe just leaving off the `sorting` prop all together? We don't really care how/if the rows are sorted for this test case, we just want to know if they're shown/not shown.

That would simplify the test + remove this "...maybe the 1st row is just sorted last?" ambiguity.